cmd/exec_test.go

```diff
@@ -1045,6 +1045,116 @@ func TestFailFlagCases(t *testing.T) {
 	}
 }
 
+func TestExecWithInvalidInputOptions(t *testing.T) {
+	tests := []struct {
+		description string
+		files       map[string]string
+		stdIn       bool
+		input       string
+		expectError bool
+		expected    string
+	}{
+		{
+			description: "path passed in as arg should not raise error",
+			files: map[string]string{
+				"files/test.json": `{"foo": 7}`,
+				"bundle/x.rego": `package system
+
+		test_fun := x {
+			x = false
+			x
+		}
+
+		undefined_test {
+			test_fun
+		}`,
+			},
+			expectError: false,
+			expected:    "",
+		},
+		{
+			description: "no paths passed in as args should raise error if --stdin-input flag not set",
+			files: map[string]string{
+				"bundle/x.rego": `package system
+
+		test_fun := x {
+			x = false
+			x
+		}
+
+		undefined_test {
+			test_fun
+		}`,
+			},
+			expectError: true,
+			expected:    "requires at least 1 path arg, or the --stdin-input flag",
+		},
+		{
+			description: "should not raise error if --stdin-input flag is set when no paths passed in as args",
+			files: map[string]string{
+				"bundle/x.rego": `package system
+
+		test_fun := x {
+			x = false
+			x
+		}
+
+		undefined_test {
+			test_fun
+		}`,
+			},
+			stdIn:       true,
+			input:       `{"foo": 7}`,
+			expectError: false,
+			expected:    "",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.description, func(t *testing.T) {
+			test.WithTempFS(tt.files, func(dir string) {
+				var buf bytes.Buffer
+				params := exec.NewParams(&buf)
+				_ = params.OutputFormat.Set("json")
+				params.BundlePaths = []string{dir + "/bundle/"}
+				if tt.stdIn {
+					params.StdIn = true
+					tempFile, err := os.CreateTemp("", "test")
+					if err != nil {
+						t.Fatalf("unexpected error creating temp file: %q", err.Error())
+					}
+					if _, err := tempFile.Write([]byte(tt.input)); err != nil {
+						t.Fatalf("unexpeced error when writing to temp file: %q", err.Error())
+					}
+					if _, err := tempFile.Seek(0, 0); err != nil {
+						t.Fatalf("unexpected error when rewinding temp file: %q", err.Error())
+					}
+					oldStdin := os.Stdin
+					defer func() {
```

Comment:
[nit][ We could use `t.Cleanup()` here. But it doesn't make a difference, I believe.